app/models/search/async_indexer.rb

```diff
@@ -14,6 +14,9 @@ def self.perform(name)
     REDIS.del(name)
   end
 
+  def self.index(klass, ids, priority)
```

Comment:
Looks like this is missing some things - I just submitted a PR to your branch to flesh that out, since I was doing some batch indexing to test things out locally.